SYCL/ESIMD/api/functional/functions/functions_1d_select.hpp

```diff
@@ -0,0 +1,281 @@
+                      " operator is not equal to the expected one for simd<" +
+                      data_type + ", " + std::to_string(NumElems) + ">.");
+    }
+
+    return passed;
+  }
+
+private:
+  bool fail_test(size_t i, DataT expected, DataT retrieved,
+                 const std::string &data_type) {
+    log::fail(TestDescription<NumElems, TestCaseT>(data_type),
+              "Unexpected value at index ", i, ", retrieved: ", retrieved,
+              ", expected: ", expected, ", with size: ", NumSelectedElems,
+              ", stride: ", Stride, ", offset: ", Offset);
+
+    return false;
+  }
+};
+
+template <tested_types TestedTypes, typename SelectT>
+bool run_test_for_types(sycl::queue &queue) {
+  bool passed = true;
+  constexpr int desired_simd_small_size = 1;
+  constexpr int desired_simd_large_size = 16;
+  constexpr int coefficient_of_division = 3;
+  constexpr int zero_offset_value = 0;
+  constexpr int small_offset_value = 1;
+  constexpr int large_offset_value =
+      desired_simd_large_size - details::ceil(2 * desired_simd_large_size / 3);
+
+  const auto small_size = get_dimensions<desired_simd_small_size>();
+  const auto great_size = get_dimensions<desired_simd_large_size>();
+  const auto types = get_tested_types<TestedTypes>();
+
+  // Checks are run for specific combinations of types, sizes, strides and
+  // offsets.
+  passed &=
+      for_all_combinations<run_test, SelectT, size_type<1>, stride_type<1>,
+                           offset_type<zero_offset_value>>(types, small_size,
+                                                           queue);
+  passed &=
+      for_all_combinations<run_test, SelectT, size_type<1>, stride_type<1>,
+                           offset_type<zero_offset_value>>(types, small_size,
+                                                           queue);
+  passed &= for_all_combinations<
+      run_test, SelectT,
+      size_type<desired_simd_large_size / coefficient_of_division>,
+      stride_type<coefficient_of_division>, offset_type<zero_offset_value>>(
+      types, great_size, queue);
+  passed &= for_all_combinations<
+      run_test, SelectT,
+      size_type<desired_simd_large_size / coefficient_of_division>,
+      stride_type<3>, offset_type<zero_offset_value>>(types, great_size, queue);
+  passed &= for_all_combinations<
+      run_test, SelectT, size_type<3>,
+      stride_type<desired_simd_large_size / coefficient_of_division>,
+      offset_type<zero_offset_value>>(types, great_size, queue);
+
+  passed &= for_all_combinations<
+      run_test, SelectT,
+      size_type<desired_simd_large_size - small_offset_value>,
+      stride_type<desired_simd_small_size>, offset_type<small_offset_value>>(
+      types, great_size, queue);
+  passed &=
+      for_all_combinations<run_test, SelectT,
+                           size_type<desired_simd_large_size / 3>,
+                           stride_type<2>, offset_type<large_offset_value>>(
+          types, great_size, queue);
```

Comment:
Actually, I would suggest at least to use something like `run_with_size_stride_offset` if it makes sense.. Maybe I'm missing something, but it's definitely hard to read and need to be refactored in any way.

Reply:
The `run_with_size_stride_offset` has been implemented in this commit: https://github.com/intel/llvm-test-suite/pull/875/commits/4fe9429145e7b85bbf4457dc385e8d749ba08d74

But I am not sure that this is a good solution